### DISCUSSION TITLE
Can you add a feature to show that if caps lock is on or off?

### DISCUSSION THREAD

**Discussion by MAXIIIMVS** [nvim-lualine/lualine.nvim#841] [Ideas]
Hi!
Is there any way to add a feature to show the status of caps lock?
I found one for tmux, but I didn't like it (it had some delay).
It would be awesome if you could add an optional feature (like an icon) that shows whether the caps lock is on or off.

**Comment by Pizza989** [comment]
https://github.com/barklan/capslock.nvim?tab=readme-ov-file#statusline-integration

this seems to be able to do what you want

**Comment by digimokan** [comment]
It looks like capslock.nvim only shows whether the capslock-nvim software caps lock is active. It doesn't show whether a hardware (key-toggled) caps lock is active.

**Comment by nikita-edel** [comment]
<https://github.com/nikita-edel/capsdetect.nvim>  
this works on windows/macos/linux, you can set a callback to update lualine every time caps changes
